airbyte-webapp/src/packages/cloud/components/experiments/SignupSourceDropdown/useGetSourceDefinitions.tsx

```diff
@@ -0,0 +1,30 @@
+import { useQuery } from "react-query";
+
+import { DestinationDefinitionRead, SourceDefinitionRead } from "core/request/AirbyteClient";
+
+import availableSourceDefinitions from "./sourceDefinitions.json";
+
+interface Catalog {
+  destinations: DestinationDefinitionRead[];
+  sources: SourceDefinitionRead[];
+}
+const fetchCatalog = async (): Promise<Catalog> => {
+  const path = "https://storage.googleapis.com/prod-airbyte-cloud-connector-metadata-service/cloud_catalog.json";
+  const response = await fetch(path);
+  return response.json();
+};
+export const useGetSourceDefinitions = () => {
+  return useQuery<Catalog, Error, Catalog["sources"]>("cloud_catalog", fetchCatalog, {
+    select: (data) => {
+      return data.sources.map((source) => {
+        const icon = availableSourceDefinitions.sourceDefinitions.find(
```

Comment:
Since the cloud_catalog does not include icons, I have a hardcoded `.json` with the sources that we have currently to get the icons. This is a hacky solution, but Pedro mentioned that having the actual svgs on the cloud catalog is not that trivial.  https://github.com/airbytehq/airbyte-cloud/issues/3261#issuecomment-1291242003

I think this is good enough for the experiment

Reply:
Given that we know, that this won't be the final solution here, and we already merge from the catalog directly from the googleapis plus your local one, I'd suggest we just remove the call to the cloud_catalog.json altogether and for the time being while this experiment runs just use an embedded JSON file instead. Given we anyway need to rework that for production use I feel it's easier (plus faster, since we don't need to do the additional request on the sign-up page).

Reply:
The benefit that I see on keeping it, is that we'll get the more recent cloud catalog. If we embed the json, we'll miss that. If that's ok (I'm not sure how often connectors change) I agree on having just the json file.